plugins/single_plugins/fast-switcher.cpp

```diff
@@ -105,12 +118,77 @@ class wayfire_fast_switcher : public wf::per_output_plugin_instance_t, public wf
 
     const std::string transformer_name = "fast-switcher";
 
-    void set_view_alpha(wayfire_view view, float alpha)
+    wf::effect_hook_t pre_hook = [this] ()
+    {
+        for (auto view : views)
+        {
+            auto anim_data = view->get_data<animation_data_t>();
+            if (!anim_data)
+            {
+                continue;
+            }
+
+            auto transformer =
+                view->get_transformed_node()->get_transformer<wf::scene::view_2d_transformer_t>(
+                    transformer_name);
+            if (!transformer)
+            {
+                continue;
+            }
+
+            transformer->alpha = static_cast<double>(anim_data->anim);
+            view->damage();
+        }
+    };
+
+    wf::effect_hook_t post_hook = [this] ()
```

Comment:
I am confused as to why we need a pre and a post hook, is it not enough to use a pre-hook only?